### PR TITLE
Improve automated tests support on Windows

### PR DESCRIPTION
**Pull request by Peilonrayz**
This fixes the majority of #298. Most of the changes should be self-explanatory.  
However, I have changed coverage to combine the output as [a 3.8 specific coverage bug](https://github.com/nedbat/coveragepy/issues/959) was causing `cover` to fail.

I have not fixed Docs.3 as I'm unsure how I should fix it, this means `docs` is still failing.

### CODE REVIEW

File: tests/test_virtualenv.py [hunk 1]
```diff
@@ -296,12 +301,16 @@ def test_create_venv_backend(make_one):
     venv.create()
 
 
-@pytest.mark.skipif(IS_WINDOWS, reason="Not testing multiple interpreters on Windows.")
 def test_create_interpreter(make_one):
-    venv, dir_ = make_one(interpreter="python3")
+    interpreter = "3.7" if IS_WINDOWS else "python3"
```
Comment (theacodes) [inline tests/test_virtualenv.py]: Can we make this work regardless of which Python version we're testing with?

**Comment by Peilonrayz** [comment]
I'm confused. As far as I know neither the Windows or non-Windows are
testing all Python versions.

I changed python3 to 3.7 as the former can't resolve to an interpreter. Not
to version pin the tests.

How would you like to proceed?

**Comment by theacodes** [comment]
I would prefer a single session (like tests-3.6) be able to pass if *only* that interpreter is installed. Does that make sense?

**Comment by Peilonrayz** [comment]
I don't understand what you mean. Are you saying if a user only has Python
3.6 installed, you want all the tests to pass?

When making the change I looked at the rest of the file and see lots of 3.x
specifiers. I don't understand why this one is important and the others are
not. Is this also a problem with `test_condaenv_create_interpreter`?

**Comment by theacodes** [comment]
Yes, the test should either pass or skip. I can't see why this test can't be changed to work with whatever python version the tests are currently running under.

**Comment by Peilonrayz** [comment]
OK, is not passing an interpreter, on just Windows, an acceptable
solution?
That way the interpreter is guaranteed to exist.

**Comment by theacodes** [comment]
sure

**Comment by theacodes** [comment]
Howdy, @Peilonrayz. It looks like this PR has gone stale. If you're still interested in getting it merged, let me know and we can get this started again.

**Comment by Peilonrayz** [comment]
Sorry life got busy. I've looked into the problem again and `make_one(interpreter="python3")` fails to resolve to an interpreter for me. Maybe I have a bug with my setup or nox has a bug making the test fail. I have gotten `make_one(interpreter=None)` to work on Windows but I'd just be adding a potential bug on Linux since the `assert dir_.join("bin", "python3").check()` could fail if Python 2 is used. I could use `make_one(interpreter=None if IS_WINDOWS else "python3")`, which smells to me.

Bottom line; I don't know how to nicely solve the issue we have here.

Would the best course of action be to just remove the changes to `test_create_interpreter` and let the rest of the PR through?

**Comment by theacodes** [comment]
> Would the best course of action be to just remove the changes to
test_create_interpreter and let the rest of the PR through?

i'm fine with that. :)

**Comment by Peilonrayz** [comment]
Ok! I've undone the changes to `test_create_interpreter`, but the commit isn't showing up here for me. This is my first time working on a closed PR; do I need to create a new PR?

**Comment by theacodes** [comment]
I re-opened it. Looks like you may need to `git push` though?

**Comment by Peilonrayz** [comment]
Thank you for reopening my PR. FWIW I edited the file on GitHub and when I got to the issue today the commit had appeared here.

I had to merge some changes, for the most part I took your changes. Where I kept `coverage combine` to fix a bug in Python that affects coverage on Windows. Since I can't reproduce the issue - you may have changed the one bit of code which had the bug. I'm happy to remove the call to coverage combine if you'd like.

**Comment by theacodes** [comment]
Thank you!!
